clients/roscpp/src/libros/spinner.cpp

```diff
@@ -70,17 +149,11 @@ MultiThreadedSpinner::MultiThreadedSpinner(uint32_t thread_count)
 
 void MultiThreadedSpinner::spin(CallbackQueue* queue)
 {
-  boost::recursive_mutex::scoped_try_lock spinlock(spinmutex);
-  if (!spinlock.owns_lock()) {
-    ROS_ERROR("MultiThreadeSpinner: You've attempted to call ros::spin "
-              "from multiple threads... "
-              "but this spinner is already multithreaded.");
-    return;
-  }
   AsyncSpinner s(thread_count_, queue);
   s.start();
 
   ros::waitForShutdown();
+  s.stop();
```

Comment:
This call is redundant and should be removed.

@rhaschke If you could update this that would be great. Otherwise I can apply it during the merge.